### PR TITLE
Change range positions to be zero-indexed instead of one-indexed

### LINKED ISSUE #49: Ranges are currently 1 index based, protocol requires zero-based indices

### PR DESCRIPTION
**Pull request by natgabb**
Closes #49

### CODE REVIEW

File: groovy-language-server/src/main/java/com/palantir/ls/groovy/GroovycWrapper.java [hunk 1]
```diff
@@ -358,4 +359,10 @@ private static void addToValueSet(Map<String, Set<SymbolInformation>> map, Strin
         map.computeIfAbsent(key, (value) -> Sets.newHashSet()).add(symbol);
     }
 
+    private static Range createZeroBasedRange(int startLine, int startChar, int endLine, int endChar) {
```
Comment (jiahuijiang) [inline groovy-language-server/src/main/java/com/palantir/ls/groovy/GroovycWrapper.java]: you can put this function into the `Ranges` util class

Comment (natgabb) [inline groovy-language-server/src/main/java/com/palantir/ls/groovy/GroovycWrapper.java]: good point, fixed!

**Comment by jiahuijiang** [comment]
+1

File: groovy-language-server/src/main/java/com/palantir/ls/util/Ranges.java [hunk 2]
```diff
@@ -50,6 +50,16 @@ public static Range createRange(int startLine, int startColumn, int endLine, int
...
 
     /**
+     * Returns a newly created zero-indexed range from one-indexed lines and columns.
+     */
+    public static Range createZeroBasedRange(int startLine, int startColumn, int endLine, int endColumn) {
+        Range range = Ranges.createRange(startLine - 1, startColumn - 1,
+                endLine - 1, endColumn - 1);
+        return Ranges.isValid(range) ? range : Ranges.UNDEFINED_RANGE;
+    }
+
```
Comment (gracew) [inline groovy-language-server/src/main/java/com/palantir/ls/util/Ranges.java:60]: extra line

**Comment by gracew** [comment]
+1

**Pull request merged**